ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13906,8 +13906,12 @@ ASTNode analyzeCreateTable(
       rootTasks.add(TaskFactory.get(new DDLWork(getInputs(), getOutputs(), crtTranTblDesc)));
       break;
 
-    case CTLT: // create table like <tbl_name>
+    case CTLT:// create table like <tbl_name>
 
+      if (handler != null && !handler.supportsCTLT()) {
+        throw new SemanticException("Unsupported operation. " + qualifiedTabName.getTable() +
+            " cannot be created using CTLT syntax. ");
```

Comment:
Should we have an entry in the ErrorMsg.java for this case. There are couple of similar instanses for unspupported cases like ``TRUNCATE_FOR_NON_MANAGED_TABLE`` & for CTAS as well ``CTAS_LOCATION_NONEMPTY`` we can we have one like ``CTLT_FOR_ICEBERG_TABLE`` for this case?

Reply:
Good idea. I will add a new error type to the class.